src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs

```diff
@@ -14,6 +15,16 @@ internal static class TensorTypeExtensions
     {
         public static void ToScalar<T>(this Tensor tensor, ref T dst) where T : unmanaged
         {
+            //In ML.NET we are using ReadOnlyMemory<Char> to store string data but ReadOnlyMemory<Char>
+            //is not valid data type for tensorflow.net and exception will thrown if we call as_dtype method
+            //so we specially deal with string type here.
+            //Get string data first then convert to ReadOnlyMemory<Char> and assign value to dst.
+            if (typeof(T).Name.Contains("ReadOnlyMemory"))
```

Comment:
Should this be `if (typeof(T) == typeof(ReadOnlyMemory<char>))`?

Reply:
Thanks, fixed

---
In reply to: [517666330](https://github.com/dotnet/machinelearning/pull/5468#discussion_r517666330) [](ancestors = 517666330)